infer/lib/python/inferlib/capture/gradle.py

```diff
@@ -62,19 +81,38 @@ def get_infer_commands(self, verbose_output):
                 javac_arguments = content.split(' ')
                 java_files = []
                 java_args = []
+                collected = []
                 for java_arg in javac_arguments:
-                    if java_arg.endswith('.java'):
-                        java_files.append(java_arg)
-                    else:
+                    if java_arg and java_arg[0] == '-':
```

Comment:
This won't work for "SomeFileWith -ASpaceAndADash.java" for instance, or for files starting with a `-`.